pkg/api/types.go

```diff
@@ -1375,8 +1375,42 @@ type BuildArg struct {
 	// Name of the build arg.
 	Name string `json:"name,omitempty"`
 
-	// Value of the build arg.
+	// Value of the build arg. Cannot be set if ValueFrom is set.
 	Value string `json:"value,omitempty"`
+
+	// ValueFrom specifies the source of the build Arg. Cannot be set if Value is set.
+	ValueFrom *BuildArgSource `json:"value_from,omitempty"`
+}
+
+// BuildArgSource represents a source for the value of a BuildArg.
+type BuildArgSource struct {
```

Comment:
I don't understand this structure. Why is the keyref nested? Why are the key and secret name correlated in a separate struct? What does it mean for this:

```yaml
value_from:
  namespace: thing
```

?

This is implied as possible by having a pointer to `SecretKeySelector` and allowing to to be `omitempty`

Reply:
Put them to the same level.
I was thinking the value could be taken from config as well in the future.